src/cljs/pyregence/pages/verify_2fa.cljs

```diff
@@ -0,0 +1,76 @@
+  (:require [cljs.reader                    :as edn]
+            [clojure.core.async             :refer [go <! timeout]]
+            [pyregence.components.common    :refer [simple-form]]
+            [pyregence.components.messaging :refer [toast-message!]]
+            [pyregence.styles               :as $]
+            [pyregence.utils.async-utils    :as u-async]
+            [pyregence.utils.browser-utils  :as u-browser]
+            [pyregence.utils.data-utils     :as u-data]
+            [reagent.core                   :as r]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; State
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defonce pending?           (r/atom false))
+(defonce email              (r/atom ""))
+(defonce verification-code  (r/atom ""))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; API Calls
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn- verify-2fa! []
+  (go
+    (reset! pending? true)
+    (let [errors (remove nil?
+                         [(when (empty? @verification-code)
+                            "Please enter the verification code.")])]
+
+      (if (pos? (count errors))
+        (do (toast-message! errors)
+            (reset! pending? false))
+        (if (:success (<! (u-async/call-clj-async! "verify-2fa" @email @verification-code)))
+          (do (toast-message! "Your verification code has been verified successfully.")
+              (<! (timeout 2000))
+              (u-browser/jump-to-url! "/forecast"))
+          (do (toast-message! "Invalid verification code. Please try again.")
+              (reset! pending? false)))))))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; UI Components
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn root-component
+  "The root component for the /verify-2fa page.
+   Displays the 2FA verification form."
+  [params]
+  (reset! email (or (:email params) ""))
+  ;; If email is missing, redirect to login page
+  (when (empty? @email)
+    (u-browser/jump-to-url! "/login"))
+  (fn [_]
+    [:<>
+     [:div {:style ($/combine ($/disabled-group @pending?)
+                              {:display "flex" :justify-content "center" :margin "5rem"})}
+      [simple-form
+       "Two-Factor Authentication"
+       "Verify"
+       [["Verification Code" verification-code "text" "verification-code"]]
+       verify-2fa!]]
+     [:div {:style {:display "flex"
+                    :flex-direction "column"
+                    :align-items "center"
+                    :margin-top "1rem"}}
+      [:p {:style {:margin-bottom "0.5rem"}}
+       (str "Enter the verification code sent to ")
+       [:span {:style {:font-weight "bold"}} @email]
+       "."]
+      [:a {:href "#"
+           :style {:color ($/color-picker :link-color)}
+           :on-click #(go
+                        (if (:success (<! (u-async/call-clj-async! "send-email" @email :2fa)))
```

Comment:
Mailing works:

![image](https://github.com/user-attachments/assets/ae37edf2-c361-4b5e-be63-147780eb8315)

Reply:
Passing wrong code works (rejects):

![image](https://github.com/user-attachments/assets/0b5f7509-2c56-4c97-a7e4-fa8090c1ee1e)

(Appears a message: "Invalid verification code. Please try again.")

Reply:
If pass the correct code, it logs in.